scm-core/src/main/java/sonia/scm/web/filter/AuthenticationFilter.java

```diff
@@ -226,30 +192,27 @@ private AuthenticationToken createToken(HttpServletRequest request)
   /**
    * Handle authentication with the given {@link AuthenticationToken}.
    *
-   *
-   * @param request http servlet request
+   * @param request  http servlet request
    * @param response http servlet response
-   * @param chain filter chain
-   * @param subject subject
-   * @param token authentication token
-   *
+   * @param chain    filter chain
+   * @param subject  subject
+   * @param token    authentication token
    * @throws IOException
    * @throws ServletException
    */
   private void handleAuthentication(HttpServletRequest request,
-    HttpServletResponse response, FilterChain chain, Subject subject,
-    AuthenticationToken token)
-    throws IOException, ServletException
-  {
+                                    HttpServletResponse response, FilterChain chain, Subject subject,
+                                    AuthenticationToken token)
+    throws IOException, ServletException {
     logger.trace("found basic authorization header, start authentication");
 
-    try
-    {
+    try {
       subject.login(token);
       processChain(request, response, chain, subject);
-    }
-    catch (AuthenticationException ex)
-    {
+    } catch (SessionExpiredException ex) {
+      logger.warn("session expired", ex);
+      handleUnauthorized(request, response, chain);
```

Comment:
Do we have to log always the whole stacktrace on warn? Is this not a normal thing?
What about something like this:

```suggestion
      if (logger.isTraceEnabled()) {
        logger.trace("{} expired", token.getClass(), ex);
      } else {
        logger.debug("{} expired", token.getClass());
      }
      handleUnauthorized(request, response, chain);
```